File: jaros/llm/adapters/default_adapter.py
"""Deterministic echo adapter implementing :class:`LlmClient` (EXT-004 / REQ-2).

A concrete, dependency-free adapter used as the default model. It is
deterministic (same request -> same response) and returns inert data only,
never driving control flow (EXT-004 / REQ-3).
"""

from __future__ import annotations

from dataclasses import dataclass

from jaros.llm.client import LlmRequest, LlmResponse


# #EXT-004-REQ-2 Start
@dataclass(frozen=True)
class DefaultAdapter:
    """Deterministic echo adapter that conforms to :class:`LlmClient`.

    Returns the prompt verbatim as ``text``. Holds no provider handles and
    performs no side effects.
    """

    model: str = "default-echo"

    def complete(self, req: LlmRequest) -> LlmResponse:
        """Echo the prompt back as data-only output."""
        return LlmResponse(text=req.prompt, model=self.model)
# #EXT-004-REQ-2 End

File: src/llm/adapters/default-adapter.ts
import type { LlmClient, LlmRequest, LlmResponse } from "../client";

// #EXT-004-REQ-2 Start
/**
 * A concrete, dependency-free `LlmClient`. This is the swappable boundary, not a
 * real provider integration, so it is a deterministic stub: it derives its
 * response purely from the request (echoes the prompt back). A real adapter
 * would call out to a provider SDK here instead.
 *
 * Per the Prime Directive ([PRIME-001 / REQ-4], EXT-004 REQ-3) the adapter
 * returns DATA ONLY. It never invokes a state transition, never performs a side
 * effect, and never hands a runtime handle back to the caller.
 */
export class DefaultAdapter implements LlmClient {
  /** Stable identifier reported on every response's `model` field. */
  static readonly model = "default-echo";

  async complete(req: LlmRequest): Promise<LlmResponse> {
    // Deterministic derivation: echo the prompt. Pure data in → data out.
    return {
      text: `echo: ${req.prompt}`,
      model: DefaultAdapter.model,
    };
  }
}
// #EXT-004-REQ-2 End
